tensorflow/compiler/xla/stream_executor/rocm/rocm_blas.cc

```diff
@@ -881,12 +881,19 @@ bool ROCMBlas::DoBlasGemmBatched(
 bool ROCMBlas::DoBlasGemmBatched(
     Stream *stream, blas::Transpose transa, blas::Transpose transb, uint64_t m,
     uint64_t n, uint64 k, float alpha,
-    const absl::Span<DeviceMemory<Eigen::bfloat16> *const> &a, int lda,
-    const absl::Span<DeviceMemory<Eigen::bfloat16> *const> &b, int ldb,
-    float beta, const absl::Span<DeviceMemory<Eigen::bfloat16> *const> &c,
-    int ldc, int batch_count, ScratchAllocator *scratch_allocator) {
-  LOG(ERROR) << "DoBlasGemmBatched not implemented for bfloat16";
-  return false;
+    const absl::Span<DeviceMemory<Eigen::bfloat16> *const> &a_array, int lda,
+    const absl::Span<DeviceMemory<Eigen::bfloat16> *const> &b_array, int ldb,
+    float beta, const absl::Span<DeviceMemory<Eigen::bfloat16> *const> &c_array,
+    int ldc, int batch_count, ScratchAllocator *scratch_allocator) {  
+  blas_log("DoBlasGemmBatched");
+  ports::Status status = DoBlasGemmBatchedInternal(
+    wrap::rocblas_gemm_ex, stream, transa, transb, m, n, k, alpha, 
```

Comment:
rocblas_gemm_ex doesn't seem to be the batched version of BlasGemm.
I also didn't find any function that could be used instead.
Can you please ask someone else from AMD whether this is the correct function?
For example in the source file gemm_strided_batched.cpp, I see the following function names:

```

    template <>
    constexpr char rocblas_gemm_strided_batched_name<rocblas_half>[]
        = "rocblas_hgemm_strided_batched";

    template <>
    constexpr char rocblas_gemm_strided_batched_name<float>[] = "rocblas_sgemm_strided_batched";

    template <>
    constexpr char rocblas_gemm_strided_batched_name<double>[] = "rocblas_dgemm_strided_batched";

    template <>
    constexpr char rocblas_gemm_strided_batched_name<rocblas_float_complex>[]
        = "rocblas_cgemm_strided_batched";

    template <>
    constexpr char rocblas_gemm_strided_batched_name<rocblas_double_complex>[]
        = "rocblas_zgemm_strided_batched";
```

There doesn't seem to be one for bfloat16.

Reply:
Thanks a lot for the checking! 
The batched version with bf16 support is from rocBLAS extention (see the following link), rocblas_gemm_batched_ex and I will correct it. https://rocblas.readthedocs.io/en/master/API_Reference_Guide.html#_CPPv423rocblas_gemm_batched_ex14rocblas_handle17rocblas_operation17rocblas_operation11rocblas_int11rocblas_int11rocblas_intPKvPKv16rocblas_datatype11rocblas_intPKv16rocblas_datatype11rocblas_intPKvPKv16rocblas_datatype11rocblas_intPv16rocblas_datatype11rocblas_int11rocblas_int16rocblas_datatype17rocblas_gemm_algo7int32_t8uint32_t

rocblas_gemm_strided_batched_name is level-3 function (https://rocblas.readthedocs.io/en/master/API_Reference_Guide.html#rocblas-xgemm-batched-strided-batched) for half, float, double, float_complext and double_complex, but rocBLAS extension (rocblas_gemm_ex) has more support.

Hope it makes sense.

Reply:
Thanks for the explanation. In that case, just the two other things (see my other comments) need to be fixed to make the code compile.

Reply:
@akuegel Sorry for previous typos. All have fixed and passed the test. Please take another look. Thanks in advance!